frontend/src/metabase/public/LocaleProvider.tsx

```diff
@@ -15,7 +15,10 @@ export const LocaleProvider = ({
     if (locale) {
       setIsLoadingLocale(true);
       setLocaleHeader(locale);
-      loadLocalization(locale).then(() => setIsLoadingLocale(false));
+      loadLocalization(locale).then(translatedObject => {
```

Comment:
This PRs basically does 3 things.
1. Uses `LocaleProvider` on static embeds/public links components.
2. Cherry pick BE change needs to translate some strings
3. Modify this file to make some more strings work.